lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -1019,16 +1015,16 @@ listTransactionsViaCLI
     :: forall t r s . (CmdResult r, HasType Port s, KnownCommand t)
     => s
     -> ApiWallet
-    -> Maybe Iso8601Time
-    -> Maybe Iso8601Time
+    -> Maybe String
+    -> Maybe String
     -> IO r
 listTransactionsViaCLI ctx wallet mTimeRangeStart mTimeRangeEnd =
     cardanoWalletCLI @t $ join
         [ ["transaction", "list"]
         , ["--port", show (ctx ^. typed @Port)]
         , [T.unpack $ wallet ^. walletId]
-        , maybe [] (\t -> ["--start", showT t]) mTimeRangeStart
-        , maybe [] (\t -> ["--end", showT t]) mTimeRangeEnd
+        , maybe [] (\t -> ["--start", t]) mTimeRangeStart
+        , maybe [] (\t -> ["--end",   t]) mTimeRangeEnd
```

Comment:
I'd still be after using just `[String]` rather than `ApiWallet -> Maybe String -> Maybe String` :sweat_smile: because:
-  we cannot provide faulty `walletId`. 
- also we're not able to provide invalid argument like `--star`...

(I know, I know... if you're developer, your eyes must hurt when you see this ;) )

Reply:
> I'd still be after using just `[String]` rather than `ApiWallet -> Maybe String -> Maybe String` sweat_smile because:
> (I know, I know... if you're developer, your eyes must hurt when you see this ;) )

It's okay, I have a pair of special developer goggles. :sweat_smile: 

No problem, I've updated the `listTransactionsViaCLI` function as requested.

However, I do still have some reservations.

Given that this function is intended to be part of a DSL, it seems we've drawn the line rather arbitrarily on which parts of the command should be freeform, and which parts should be hard-coded. If we allow the arguments to be `[String]` (thus allowing invalid variations), then why hard-code the `transaction list` command prefix (thus disallowing invalid variations like `transaction lost` and so on)?

> also we're not able to provide invalid argument like `--star`...

I would question whether we really want to be building integration tests that check misspelled CLI arguments are rejected. It seems like there would be countless permutations, especially when we take all commands and argument combinations into account. Surely in this case, it would be better to generate such tests automatically from the CLI definition?

Reply:
>Given that this function is intended to be part of a DSL, it seems we've drawn the line rather arbitrarily on which parts of the command should be freeform, and which parts should be hard-coded. If we allow the >arguments to be [String] (thus allowing invalid variations), then why hard-code the transaction list command prefix (thus disallowing invalid variations like transaction lost and so on)?

Indeed, it is like just a shortcut to test different cases of `transaction list`. Also there is actually a _special_ test case that roughly covers cases like `transaction lost` -> https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/integration/Test/Integration/Scenario/CLI/Miscellaneous.hs#L38-L102, so it will surely land there.

>I would question whether we really want to be building integration tests that check misspelled CLI arguments are rejected. It seems like there would be countless permutations, especially when we take all commands and argument combinations into account.

My opinion is that in integration test it is not necessary about providing all possible permutations but rather examples focusing on bonduary values and corner cases and having in mind equivalence partitioning. So `--star` and `--startt` would be probably enough to test misspelled `--start`, 
 - they are sort of boundary values and kind of corner(ish) (very similar to `--start` yet different in length), 
 - Also I might assume that if they give expected result than pretty much other misspells will, so I don't need to test all possible combinations (equivalence partitioning) [well, for sanity I would probably test also some weird strings ;)]

Btw. another advantage of `[String]` is also that we can test different order of parameters (`--start ... --end ...` or `--end ... --start ...`)


>Surely, if we decide that it's valuable to test for misspellings, wouldn't it be better to generate such tests automatically from the CLI definition? (as part of a unit test suite.)

I think that testing misspellings on integration level is probably good enough, but perhaps that would be also a nice option.